PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'onboarding.onboarding.step'
-    _inherit = 'res.company'
-    _inherit = 'res.country'
-    _inherit = 'res.partner'

Total Changes: 16

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/payment/models/ir_http.py
+++ b/addons/payment/models/ir_http.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = 'ir.http'
+    _inherit = ['ir.http']
 
     @classmethod
     def _get_translation_frontend_modules_name(cls):

--- a/addons/payment/models/onboarding_step.py
+++ b/addons/payment/models/onboarding_step.py
@@ -3,8 +3,8 @@
 from odoo import api, models
 
 
-class OnboardingStep(models.Model):
-    _inherit = 'onboarding.onboarding.step'
+class OnboardingOnboardingStep(models.Model):
+    _inherit = ['onboarding.onboarding.step']
 
     @api.model
     def action_validate_step_payment_provider(self):

--- a/addons/payment/models/payment_method.py
+++ b/addons/payment/models/payment_method.py
@@ -9,7 +9,6 @@ from odoo.addons.payment.const import REPORT_REASONS_MAPPING
 
 
 class PaymentMethod(models.Model):
-    _name = 'payment.method'
     _description = "Payment Method"
     _order = 'active desc, sequence, name'
 

--- a/addons/payment/models/payment_provider.py
+++ b/addons/payment/models/payment_provider.py
@@ -12,7 +12,6 @@ _logger = logging.getLogger(__name__)
 
 
 class PaymentProvider(models.Model):
-    _name = 'payment.provider'
     _description = 'Payment Provider'
     _order = 'module_state, state desc, sequence, name'
     _check_company_auto = True

--- a/addons/payment/models/payment_token.py
+++ b/addons/payment/models/payment_token.py
@@ -7,7 +7,6 @@ from odoo.exceptions import UserError, ValidationError
 
 
 class PaymentToken(models.Model):
-    _name = 'payment.token'
     _order = 'partner_id, id desc'
     _description = 'Payment Token'
     _check_company_auto = True

--- a/addons/payment/models/payment_transaction.py
+++ b/addons/payment/models/payment_transaction.py
@@ -20,7 +20,6 @@ _logger = logging.getLogger(__name__)
 
 
 class PaymentTransaction(models.Model):
-    _name = 'payment.transaction'
     _description = 'Payment Transaction'
     _order = 'id desc'
     _rec_name = 'reference'

--- a/addons/payment/models/res_company.py
+++ b/addons/payment/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = 'res.company'
+    _inherit = ['res.company']
 
     payment_onboarding_payment_method = fields.Selection(
         string="Selected onboarding payment method",

--- a/addons/payment/models/res_country.py
+++ b/addons/payment/models/res_country.py
@@ -5,7 +5,7 @@ import odoo.addons.payment_stripe as stripe  # prevent circular import error wit
 
 
 class ResCountry(models.Model):
-    _inherit = 'res.country'
+    _inherit = ['res.country']
 
     is_stripe_supported_country = fields.Boolean(compute='_compute_is_stripe_supported_country')
 

--- a/addons/payment/models/res_partner.py
+++ b/addons/payment/models/res_partner.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     payment_token_ids = fields.One2many(
         string="Payment Tokens", comodel_name='payment.token', inverse_name='partner_id')
